src/strings.cpp

```diff
@@ -1281,10 +1289,12 @@ static char *FormatString(char *buff, const char *str_arg, StringParameters *arg
 
 			case SCC_VELOCITY: { // {VELOCITY}
 				assert(_settings_game.locale.units_velocity < lengthof(_units_velocity));
-				unsigned int decimal_places = _units_velocity[_settings_game.locale.units_velocity].decimal_places;
-				uint64 args_array[] = {ConvertKmhishSpeedToDisplaySpeed(args->GetInt64(SCC_VELOCITY)), decimal_places};
+				int64 a = args->GetInt64(SCC_VELOCITY);
+				int units = GetVelocityUnits(static_cast<VehicleType>(GB(a, 56, 8)));
```

Comment:
I for one could use an explanation of these magic numbers (just a comment is fine)